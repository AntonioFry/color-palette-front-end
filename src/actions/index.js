export const throwError = error => ({
  type: 'THROW_ERROR',
  payload: {
    error
  }
});

export const isLoading = isLoading => ({
  type: 'IS_LOADING',
  payload: {
    isLoading
  }
});

export const search = searchWord => ({
  type: 'SEARCH',
  payload: {
    searchWord
  }
});

export const addProject = project => ({
  type: 'ADD_PROJECT',
  payload: {
    project
  }
});

export const addPalette = palette => ({
  type: 'ADD_PALETTE',
  payload: {
    palette
  }
});

export const addProjects = projects => ({
  type: 'ADD_PROJECTS',
  payload: {
    projects
  }
});

export const addPalettes = palettes => ({
  type: 'ADD_PALETTES',
  payload: {
    palettes 
  }
});

export const removeProject = id => ({
  type: 'REMOVE_PROJECT',
  payload: {
    id
  }
});

export const removePalette = id => ({
  type: 'REMOVE_PALETTE',
  payload: {
    id
  }
});