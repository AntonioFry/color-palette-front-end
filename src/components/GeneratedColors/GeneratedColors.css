.color-cards-container {
  display: flex;
  width: 100%;
  height: 400px;
  flex-direction: row;
}

.generate-colors-button {
  width: 20%;
  height: 40px;
  margin: 10px 40%;
  border: 1px solid black;
  border-radius: 30px;
  font-size: 1em;
}

.generated-colors-section {
  margin: 10px 0px;
}