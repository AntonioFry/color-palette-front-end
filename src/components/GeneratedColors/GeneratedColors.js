import React, { Component } from 'react'
import ColorCard from '../ColorCard/ColorCard';

export default class GeneratedColors extends Component {

createCard = () => {
  for(var i = 0; i < this.props.palettes.length; i++){
    this.props.palettes.map(palette => {
    return <ColorCard palette={`palette.color_${i}`}/>;
  });
  }
}

  render() {
    return (
      <div>
        {this.createCard()}
      </div>
    )
  }
}
