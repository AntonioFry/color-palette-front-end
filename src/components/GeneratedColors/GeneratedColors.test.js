import React from 'react';
import { shallow } from 'enzyme';
import { GeneratedColors } from './GeneratedColors';
import { setCurrentColors } from '../../actions';

describe('GeneratedColors', () => {

  describe('GeneratedColors component', () => {
    let wrapper;
    let props;

    beforeEach(() => {
      props = {
        palettes: [
          {}
        ],
        currentColors: [
          {}
        ],
        setCurrentColors: jest.fn()
      }
      wrapper = shallow(<GeneratedColors
        currentColors={props.currentColors}
        palettes={props.palettes}
        setCurrentColors={props.setCurrentColors}
      />)
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    })
  })

})