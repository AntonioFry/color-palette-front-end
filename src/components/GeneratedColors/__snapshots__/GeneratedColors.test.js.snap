// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneratedColors GeneratedColors component should match the snapshot 1`] = `
<section
  className="generated-colors-section"
>
  <div
    className="color-cards-container"
  >
    <Connect(ColorCard) />
  </div>
  <button
    className="generate-colors-button"
    onClick={[Function]}
  >
    Generate New Palette
  </button>
</section>
`;
